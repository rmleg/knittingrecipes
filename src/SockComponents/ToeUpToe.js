import React from "react";

function ToeUpToe({ fullStCount }) {
  return (
    <div>
      <h3>Toe</h3>
      <p>
        Using{" "}
        <a
          href="http://knitty.com/ISSUEspring06/FEATmagiccaston.html"
          target="_blank"
          rel="noopener noreferrer"
        >
          Judy's Magic Cast On
        </a>
        , cast on{" "}
        <span data-testid="toeup-startingStCount">
          {fullStCount / 2 - 8 - ((fullStCount / 2 - 8) % 4)}
        </span>{" "}
        sts.
      </p>
      <p>
        Rnd 1: kfb, k to 2 sts before end of first needle, kfb, k1, kfb, k to 2
        sts before end of second needle, kfb, k1
        <br />
        Rnd 2: k around
      </p>
      <p>
        Rep rnds 1 and 2 until you have{" "}
        <span data-testid="toeup-finalStCount">{fullStCount}</span> sts.{" "}
        <em>
          For a slightly more rounded toe, try repeating the first rnd twice
          before starting to repeat both rnds.
        </em>
      </p>
    </div>
  );
}

export default ToeUpToe;
