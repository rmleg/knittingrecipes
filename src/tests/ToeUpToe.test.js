import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import ToeUpToe from "../SockComponents/ToeUpToe";

afterEach(cleanup);

test("Toe up toe should render correctly", () => {
  const { container } = render(<ToeUpToe fullStCount="60" />);
  expect(container.firstChild).toMatchSnapshot();
});

test("Difference between toe up starting st count and final st count should be a multiple of 4", () => {
  render(<ToeUpToe fullStCount="60" />);
  let startingStCount = Number(
    screen.getByTestId("toeup-startingStCount").textContent
  );
  let fullStCount = Number(
    screen.getByTestId("toeup-finalStCount").textContent
  );
  expect((fullStCount - startingStCount) % 4).toBe(0);
  cleanup();

  render(<ToeUpToe fullStCount="64" />);
  startingStCount = Number(
    screen.getByTestId("toeup-startingStCount").textContent
  );
  fullStCount = Number(screen.getByTestId("toeup-finalStCount").textContent);
  expect((fullStCount - startingStCount) % 4).toBe(0);
  cleanup();
});
