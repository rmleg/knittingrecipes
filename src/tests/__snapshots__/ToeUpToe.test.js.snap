// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toe up toe should render correctly 1`] = `
<div>
  <h3>
    Toe
  </h3>
  <p>
    Using
     
    <a
      href="http://knitty.com/ISSUEspring06/FEATmagiccaston.html"
      rel="noopener noreferrer"
      target="_blank"
    >
      Judy's Magic Cast On
    </a>
    , cast on
     
    <span
      data-testid="toeup-startingStCount"
    >
      20
    </span>
     
    sts.
  </p>
  <p>
    Rnd 1: kfb, k to 2 sts before end of first needle, kfb, k1, kfb, k to 2 sts before end of second needle, kfb, k1
    <br />
    Rnd 2: k around
  </p>
  <p>
    Rep rnds 1 and 2 until you have
     
    <span
      data-testid="toeup-finalStCount"
    >
      60
    </span>
     sts.
     
    <em>
      For a slightly more rounded toe, try repeating the first rnd twice before starting to repeat both rnds.
    </em>
  </p>
</div>
`;
